grails-app/services/org/pih/warehouse/order/OrderSummaryService.groovy

```diff
@@ -0,0 +1,441 @@
+                                    `order`.id AS order_id,
+                                    order_adjustment.id AS order_adjustment_id,
+                                    SUM(
+                                        DISTINCT ABS(
+                                            IFNULL(
+                                                order_adjustment.amount, IF(
+                                                order_adjustment.percentage IS NOT NULL,
+                                                IF(
+                                                    order_item_id IS NOT NULL,
+                                                    order_item.quantity * order_item.unit_price * (order_adjustment.percentage / 100),
+                                                    order_total.order_total * (order_adjustment.percentage / 100)), 0)
+                                            )
+                                        )
+                                    ) AS total_adjustment
+                                FROM `order`
+                                    LEFT OUTER JOIN order_adjustment ON order_adjustment.order_id = `order`.id
+                                    LEFT OUTER JOIN (${getOrderAdjustmentPaymentStatusSelect(orderId)}) order_adjustment_payment_status ON order_adjustment_payment_status.adjustment_id = order_adjustment.id
+                                    LEFT OUTER JOIN order_item ON order_adjustment.order_item_id = order_item.id
+                                    LEFT OUTER JOIN (
+                                        SELECT oi.order_id AS order_id,
+                                            SUM(oi.quantity * oi.unit_price) AS order_total
+                                        FROM order_item oi
+                                            LEFT OUTER JOIN `order` o ON o.id = oi.order_id
+                                        GROUP BY oi.order_id
+                                    ) AS order_total ON `order`.id = order_total.order_id
+                                WHERE order_adjustment.canceled IS NOT TRUE
+                                GROUP BY order_adjustment_id, order_id
+                            ) AS amount_per_order_by_adjustment
+                        GROUP BY amount_per_order_by_adjustment.order_id) AS total_adjustments ON total_adjustments.order_id = `order`.id
+                    WHERE `order`.order_type_id = 'PURCHASE_ORDER'
+                        AND `order`.id = '${orderId}'
+                        AND order_adjustment.canceled IS NOT TRUE
+                    GROUP BY `order`.id) AS items_and_adjustments_union
+                GROUP BY id, order_status) AS order_summary
+            WHERE id = '${orderId}'
+        """
+    }
+
+    /**
+     * Refreshing entire Order Summary materialized view, should be only triggered from time to time
+     * (same statements as in the order-summary-materialized-view.sql)
+     * */
+    def refreshOrderSummary() {
```

Comment:
```suggestion
    void refreshOrderSummary() {
```

Reply:
the same change in the functions below

Reply:
I just moved these few refreshes, from order service, but this makes sens